### PR TITLE
Protecting lambda functions, in case cuda is enabled.

### PR DESCRIPTION
**Pull request by mndevec**
Should solve #72

### CODE REVIEW

File: src/common/KokkosKernels_SparseUtils.hpp [hunk 1]
```diff
@@ -1582,6 +1581,7 @@ void kk_create_incidence_tranpose_matrix_from_lower_triangle(
     bool use_dynamic_scheduling = false,
     bool chunksize = 4){
 
+#ifndef KOKKOS_HAVE_CUDA
```
Comment (crtrott) [inline src/common/KokkosKernels_SparseUtils.hpp]: Rename to KOKKOS_ENABLE_CUDA

**Comment by mndevec** [comment]
@mhoemmen 
It is basically disabling the code that is written with lambdas when Cuda is enabled. This code is not used in any downstream.

**Comment by mndevec** [comment]
I have changed the use of KOKKOS_HAVE_CUDA to KOKKOS_ENABLE_CUDA all over the package. 
We will need to the same of OpenMP, Serial, and Pthreads. 
@srajama1 @kyungjoo-kim  @ambrad

File: perf_test/graph/KokkosGraph_triangle.cpp [hunk 2]
```diff
@@ -358,7 +358,7 @@ int main (int argc, char ** argv){
 #endif
 
 
-#if defined( KOKKOS_HAVE_CUDA1 )
+#if defined( KOKKOS_ENABLE_CUDA1 )
```
Comment (mhoemmen) [inline perf_test/graph/KokkosGraph_triangle.cpp]: What's "CUDA1"?  Did you mean to type `KOKKOS_ENABLE_CUDA` instead?

**Comment by mhoemmen** [comment]
@mndevec Argh, I should have looked harder: what is `KOKKOS_ENABLE_CUDA1`?  Did you mean to type `KOKKOS_ENABLE_CUDA` instead?

File: perf_test/graph/KokkosGraph_triangle.cpp (hunk 2 above)
Comment (crtrott) [reply perf_test/graph/KokkosGraph_triangle.cpp]: Jup thats a typo we overlooked ...

**Comment by mndevec** [comment]
@mhoemmen 
No this code is disabled in purpose. I need to make couple more optimizations to run the code on GPUs. 
For example, it is currently using parallel GNU sort.

File: perf_test/graph/KokkosGraph_triangle.cpp (hunk 2 above)
Comment (mndevec) [reply perf_test/graph/KokkosGraph_triangle.cpp]: It is not a typo, it is intentional to disable the code.

**Pull request merged**